docs/io/ascii/read.rst

```diff
@@ -740,6 +740,57 @@ the methods in the base class::
       header_class = CommentedHeaderHeader
       data_class = NoHeaderData
 
+**Application: Write a "fixed_width" table with a "commented_header"**
+
+This module provides formats for tables where the header line is marked with a comment
+character and a separate class that writes fixed-width tables, but there is no functionality
+to write a fixed-width table with a commented header. Fixed width tables can be easier to read
+by eye because the rows are aligned and certain other programs require the header line to be
+commented. So, we now want to make a writer that can write this format; for this example we do
+not bother to work out how to read this format, but just raise an error on reading:
+
+  >>> from astropy.io.ascii.fixedwidth import FixedWidthData, FixedWidth
+  >>> class FixedWidthDataCommentedHeader(FixedWidthData):
```

Comment:
The naming convention for the `Data` reader subclass is `<ReaderClass>Data`. So this should be `FixedWidthCommentedHeaderData`.